src/simulator/distance_measuring_sensor.py

```diff
@@ -13,26 +13,29 @@ class DistanceMeasuringSensor(sensor.Sensor):
     it is a sensor that measures the distance of an object to where the sensor is located.
     """
     def __int__(self, oid: object.ObjectId, name: str, pos: position.Position,
-                owner_object: object.Object, registry: object_registry.ObjectRegistry, parameter: float):
+                owner_object: object.Object, registry: object_registry.ObjectRegistry, sensor_range: float):
         self.position = pos
         new_position = copy.copy(self.position)
```

Comment:
1. you're taking the phi of the beam as that of the sensor. This is not good. will cause problems. Add beam_phi and beam_theta to the arguments of the constructor. So that we can define the beam as flexibly as we'd like.
2. following the above comment, no need to make a copy. Just instantiate a new Position